src/datatype/svecf32.rs

```diff
@@ -0,0 +1,326 @@
+use crate::prelude::*;
+use pgrx::pg_sys::Datum;
+use pgrx::pg_sys::Oid;
+use pgrx::pgrx_sql_entity_graph::metadata::ArgumentError;
+use pgrx::pgrx_sql_entity_graph::metadata::Returns;
+use pgrx::pgrx_sql_entity_graph::metadata::ReturnsError;
+use pgrx::pgrx_sql_entity_graph::metadata::SqlMapping;
+use pgrx::pgrx_sql_entity_graph::metadata::SqlTranslatable;
+use pgrx::FromDatum;
+use pgrx::IntoDatum;
+use service::prelude::*;
+use std::alloc::Layout;
+use std::cmp::Ordering;
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::ops::Deref;
+use std::ops::DerefMut;
+use std::ops::Index;
+use std::ops::IndexMut;
+use std::ptr::NonNull;
+
+#[repr(C, align(8))]
+pub struct SVecf32 {
+    varlena: u32,
+    len: u16,
+    kind: u8,
+    reserved: u8,
+    dims: u16,
```

Comment:
There is a pair of padding bytes and we need to fill them with zeros. It's required by PostgreSQL.

Reply:
completed in new commit